### PR TITLE
Update infinispan to version 12.1.7

### PR DESCRIPTION
**Pull request by anjumfatima90**

### CODE REVIEW

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

**Comment by zhang-ca** [comment]
The failures were from EE10_FEATURES only:

```
[3/5/24, 23:23:54:337 PST] 0000002c id=00000000 com.ibm.ws.session.store.cache.CacheHashMap                  3 ==> CachingProvider73d2f363.getCacheManager null, null, {

infinispan.client.hotrod.auth_realm=default, 
infinispan.client.hotrod.auth_password=pass, 
infinispan.client.hotrod.sasl_mechanism=PLAIN, 
infinispan.client.hotrod.cache.[com.ibm.ws.session.*].template_name=org.infinispan.REPL_SYNC, 
infinispan.client.hotrod.auth_username=user, 
infinispan.client.hotrod.server_list=10.51.13.102:42527
}
[3/5/24, 23:23:54:410 PST] 0000002d id=00000000 com.ibm.ws.webcontainer.osgi.mbeans.PluginGenerator          I SRVE9103I: A configuration file for a web server plugin was automatically generated for this server at /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.session.cache.fat.infinispan.container.server/logs/state/plugin-cfg.xml.
[3/5/24, 23:23:54:833 PST] 0000002c id=00000000 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.ExceptionInInitializerError com.ibm.ws.session.store.cache.CacheStoreService 240" at ffdc_24.03.05_23.23.54.0.log
[3/5/24, 23:23:54:839 PST] 0000002c id=00000000 com.ibm.ws.session.store.cache.CacheHashMap                  3 ==> CachingProvider73d2f363.close 
[3/5/24, 23:23:54:845 PST] 0000002c id=00000000 com.ibm.ws.session.store.cache.CacheHashMap                  3 <== CachingProvider73d2f363.close 
[3/5/24, 23:23:54:845 PST] 0000002c id=00000000 com.ibm.ws.webcontainer.internal.WebContainer                E getSessionContext SRVE8059E: An unexpected exception occurred when trying to retrieve the session context
java.lang.ExceptionInInitializerError
	at org.infinispan.jcache.remote.JCacheManager.<init>(JCacheManager.java:43)
	at org.infinispan.jcache.remote.JCachingProvider.createCacheManager(JCachingProvider.java:37)
	at org.infinispan.jcache.AbstractJCachingProvider.getCacheManager(AbstractJCachingProvider.java:66)
	at com.ibm.ws.session.store.cache.CacheStoreService.lambda$activateLazily$0(CacheStoreService.java:216)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
	at com.ibm.ws.session.store.cache.CacheStoreService.activateLazily(CacheStoreService.java:171)
	at com.ibm.ws.session.store.cache.CacheHashMap.cacheInit(CacheHashMap.java:152)
	at com.ibm.ws.session.store.cache.CacheHashMap.lambda$new$0(CacheHashMap.java:138)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
	at com.ibm.ws.session.store.cache.CacheHashMap.lambda$new$1(CacheHashMap.java:138)
	at io.openliberty.checkpoint.spi.CheckpointPhase.onRestore(CheckpointPhase.java:318)
	at com.ibm.ws.session.store.cache.CacheHashMap.<init>(CacheHashMap.java:138)
```

**Comment by zhang-ca** [comment]
There was a missing class WildFlyElytronSaslPlainProvider:

```
Caused by: org.infinispan.commons.CacheConfigurationException: Unable to instantiate class org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider
	at org.infinispan.commons.util.Util.loadClass(Util.java:140)
	at org.infinispan.commons.util.Util.getInstance(Util.java:305)
	at org.infinispan.client.hotrod.RemoteCacheManager.<clinit>(RemoteCacheManager.java:138)
	... 42 more
Caused by: java.lang.ClassNotFoundException: org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider
	at com.ibm.ws.classloading.internal.UnifiedClassLoader.findClass(UnifiedClassLoader.java:167)
	at com.ibm.ws.classloading.internal.ThreadContextClassLoader.findClass(ThreadContextClassLoader.java:145)
	at com.ibm.ws.classloading.internal.UnifiedClassLoader.loadClass0(UnifiedClassLoader.java:141)
	at com.ibm.ws.classloading.internal.UnifiedClassLoader$Delegation.loadClass(UnifiedClassLoader.java:87)
	at com.ibm.ws.classloading.internal.UnifiedClassLoader.loadClass(UnifiedClassLoader.java:106)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:534)
	at java.base/java.lang.Class.forName(Class.java:513)
	at org.infinispan.commons.util.Util.loadClassStrict(Util.java:173)
	at org.infinispan.commons.util.Util.loadClass(Util.java:138)
```

File: dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle [hunk 1]
```diff
@@ -25,14 +25,16 @@ configurations {
...
   gson project(':io.openliberty.com.google.gson')
-  infinispanCommons 'org.infinispan:infinispan-commons:10.0.0.Final'
-  infinispanCore 'org.infinispan:infinispan-core:10.0.0.Final'
-  infinispanJCache 'org.infinispan:infinispan-jcache:10.0.0.Final'
-  infinispanJCacheCommons 'org.infinispan:infinispan-jcache-commons:10.0.0.Final'
+  infinispanCommons 'org.infinispan:infinispan-commons:12.1.7.Final'
+  infinispanCore 'org.infinispan:infinispan-core:12.1.7.Final'
+  infinispanJCache 'org.infinispan:infinispan-jcache:12.1.7.Final'
+  infinispanJCacheCommons 'org.infinispan:infinispan-jcache-commons:12.1.7.Final'
   protostream 'org.infinispan.protostream:protostream:4.3.0.Alpha13'
```
Comment (zhang-ca) [inline dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle]:
```suggestion
  protostream 'org.infinispan.protostream:protostream:4.4.1.Final'
```

File: dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle [hunk 2]
```diff
@@ -25,14 +25,16 @@ configurations {
...
   jbossLogging 'org.jboss.logging:jboss-logging:3.4.1.Final'
   jGroups 'org.jgroups:jgroups:4.1.6.Final'
```
Comment (zhang-ca) [inline dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle]:
```suggestion
  jGroups 'org.jgroups:jgroups:4.2.12.Final'
```

Comment (zhang-ca) [reply dev/com.ibm.ws.session.cache_fat_infinispan/build.gradle]: Test pass with above 2 changes.

File: dev/com.ibm.ws.session.cache_fat_infinispan_container/build.gradle [hunk 3]
```diff
@@ -17,14 +17,17 @@ configurations {
...
-                 'org.infinispan:infinispan-commons:10.0.0.Final',
-                 'org.infinispan:infinispan-jboss-marshalling:10.0.0.Final',
-                 'org.infinispan:infinispan-jcache-commons:10.0.0.Final',
-                 'org.infinispan:infinispan-jcache-remote:10.0.0.Final',
+                 'org.infinispan:infinispan-client-hotrod:12.1.7.Final',
+                 'org.infinispan:infinispan-commons:12.1.7.Final',
+                 'org.infinispan:infinispan-jboss-marshalling:12.1.7.Final',
+                 'org.infinispan:infinispan-jcache-commons:12.1.7.Final',
+                 'org.infinispan:infinispan-jcache-remote:12.1.7.Final',
+                 'org.infinispan.protostream:protostream:4.3.0.Alpha13',
```
Comment (zhang-ca) [inline dev/com.ibm.ws.session.cache_fat_infinispan_container/build.gradle]:
```suggestion
                 'org.infinispan.protostream:protostream:4.4.1.Final',
```

**Review by zhang-ca — COMMENTED** [review]
3 suggestions

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

File: dev/com.ibm.ws.session.cache_fat_infinispan_container/build.gradle [hunk 4]
```diff
@@ -33,8 +36,8 @@ infinispanClient 'com.github.ben-manes.caffeine:caffeine:2.8.0',
                  'io.netty:netty-handler:4.1.41.Final',
                  'io.netty:netty-resolver:4.1.41.Final',
                  'io.netty:netty-transport:4.1.41.Final',
-                 'org.wildfly.security:wildfly-elytron-base:1.10.3.Final',
-                 'org.wildfly.security:wildfly-elytron-sasl:1.10.3.Final'
+                 'org.wildfly.security:wildfly-elytron-base:1.15.1.Final',
+                 'org.wildfly.security:wildfly-elytron-sasl:1.15.1.Final'
```
Comment (zhang-ca) [inline dev/com.ibm.ws.session.cache_fat_infinispan_container/build.gradle]:
```suggestion
                 'org.wildfly.security:wildfly-elytron-sasl-plain:1.15.1.Final'
```

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

File: dev/com.ibm.ws.session.cache_fat_infinispan_container/build.gradle [hunk 5]
```diff
@@ -17,24 +17,36 @@ configurations {
...
-                 'io.netty:netty-resolver:4.1.41.Final',
-                 'io.netty:netty-transport:4.1.41.Final',
-                 'org.wildfly.security:wildfly-elytron-base:1.10.3.Final',
-                 'org.wildfly.security:wildfly-elytron-sasl:1.10.3.Final'
+                 'io.netty:netty-buffer:4.1.63.Final',
+                 'io.netty:netty-codec:4.1.63.Final',
+                 'io.netty:netty-common:4.1.63.Final',
+                 'io.netty:netty-handler:4.1.63.Final',
+                 'io.netty:netty-resolver:4.1.63.Final',
+                 'io.netty:netty-transport:4.1.63.Final',
```
Comment (zhang-ca) [inline dev/com.ibm.ws.session.cache_fat_infinispan_container/build.gradle]:
```suggestion
                 'io.netty:netty-transport-native-epoll:4.1.63.Final',
```

**Comment by zhang-ca** [comment]
#build

**Review by zhang-ca — COMMENTED** [review]
need transport jar

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
```
[4/11/24, 2:52:18:879 PDT] 0000002e id=00000000 infinispan.client.hotrod.impl.transport.netty.EPollAvailable I useNativeEpoll ISPN004074: Native Epoll transport not available, using NIO instead: java.lang.NoClassDefFoundError: io/netty/channel/unix/Errors$NativeIoException
[4/11/24, 2:52:19:177 PDT] 00000037 id=00000000 io.netty.channel.ChannelInitializer                          W exceptionCaught Failed to initialize a channel. Closing: [id: 0xacdf867e]
java.lang.NoClassDefFoundError: org/wildfly/security/sasl/util/AbstractSaslFactory
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1027)
	at com.ibm.ws.classloading.internal.AppClassLoader.definePackageAndClass(AppClassLoader.java:435)


Caused by: java.lang.ClassNotFoundException: org.wildfly.security.sasl.util.AbstractSaslFactory
	at com.ibm.ws.classloading.internal.AppClassLoader.findClassCommonLibraryClassLoaders(AppClassLoader.java:751)
	at com.ibm.ws.classloading.internal.AppClassLoader.findClass(AppClassLoader.java:327)
	at com.ibm.ws.classloading.internal.AppClassLoader.findOrDelegateLoadClass(AppClassLoader.java:705)
	at com.ibm.ws.classloading.internal.AppClassLoader.loadClass(AppClassLoader.java:584)
	at com.ibm.ws.classloading.internal.AppClassLoader.loadClass(AppClassLoader.java:553)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
```

**Comment by zhang-ca** [comment]
#build

**Review by zhang-ca — COMMENTED** [review]
Update fat_infinispan for now

**Review by zhang-ca — APPROVED** [review]

**Comment by zhang-ca** [comment]
#run-libby-bot

**Comment by zhang-ca** [comment]
#build

**Pull request merged**